backend/app/api/table_schemas.py

```
@@ -0,0 +1,109 @@
+        )
+        
+    except NotFoundError as e:
+        logger.error(f"No tables found: {e}")
+        raise HTTPException(
+            status_code=404,
+            detail={
+                "error_code": e.error_code,
+                "message": e.message,
+                "detail": e.detail
+            }
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)